memconn_pipe_go110.go

```diff
@@ -14,3 +14,12 @@ import (
 func Pipe() (net.Conn, net.Conn) {
 	return net.Pipe()
 }
+
+func isClosedChan(c <-chan struct{}) bool {
```

Comment:
Could use `context.Context` instead of the bare chan, then `ctx.Done()` instead of `isClosedChan`.  Might also be useful for callers to make use of Context to set deadlines, cancel, etc. as it is for a tcp conn.

Reply:
Hi @dougm,

Yep. I actually started down that path the first time. I walked it back because:

1. Keep it simple, keep it fast
2. I'm trying to make `memconn` at first satisfy `net.Conn`, not necessarily all the features of something like `TCPConn`, etc.

In some future update I'll likely add a custom `Dialer` like `net.Dialer` that recognizes Memconn networks and implement `context` then since the `net.Dialer` uses the context.